repairnator/repairnator-scanner/src/test/java/fr/inria/spirals/repairnator/scanner/ProjectScannerTest.java

```diff
@@ -0,0 +1,158 @@
+package fr.inria.spirals.repairnator.scanner;
+
+import ch.qos.logback.classic.Level;
+import fr.inria.jtravis.entities.Build;
+import fr.inria.jtravis.entities.StateType;
+import fr.inria.spirals.repairnator.BuildToBeInspected;
+import fr.inria.spirals.repairnator.Utils;
+import fr.inria.spirals.repairnator.config.RepairnatorConfig;
+import fr.inria.spirals.repairnator.states.LauncherMode;
+import fr.inria.spirals.repairnator.states.ScannedBuildStatus;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ProjectScannerTest {
+
+    @Before
+    public void setUp() {
+        Utils.setLoggersLevel(Level.DEBUG);
+    }
+
+    @Test
+    public void testTestBuildWithFailingWithoutFailingTestAndRepairMode() {
+        int buildId = 230022061; // inria/spoon failing from PR without failing tests
+        RepairnatorConfig config = RepairnatorConfig.getInstance();
+        config.setLauncherMode(LauncherMode.REPAIR);
+
+        Optional<Build> buildOptional = config.getJTravis().build().fromId(buildId);
+        assertTrue(buildOptional.isPresent());
+
+        Build build = buildOptional.get();
+        ProjectScanner projectScanner = new ProjectScanner(new Date(), new Date(), "test");
+        boolean isOKForRepair = projectScanner.testBuild(build);
+
+        assertFalse(isOKForRepair);
+    }
+
+    @Test
+    public void testTestBuildWithFailingTestAndRepairMode() {
+        int buildId = 364156914; // inria/spoon failing from PR with failing tests
+        RepairnatorConfig config = RepairnatorConfig.getInstance();
+        config.setLauncherMode(LauncherMode.REPAIR);
+
+        Optional<Build> buildOptional = config.getJTravis().build().fromId(buildId);
+        assertTrue(buildOptional.isPresent());
+
+        Build build = buildOptional.get();
+        ProjectScanner projectScanner = new ProjectScanner(new Date(), new Date(), "test");
+        boolean isOKForRepair = projectScanner.testBuild(build);
+
+        assertTrue(isOKForRepair);
+    }
+
+    @Test
+    public void testTestBuildWithFailingAndBearsMode() {
+        int buildId = 230022061; // inria/spoon failing from PR without failing tests
+        RepairnatorConfig config = RepairnatorConfig.getInstance();
+        config.setLauncherMode(LauncherMode.BEARS);
+
+        Optional<Build> buildOptional = config.getJTravis().build().fromId(buildId);
+        assertTrue(buildOptional.isPresent());
+
+        Build build = buildOptional.get();
+        ProjectScanner projectScanner = new ProjectScanner(new Date(), new Date(), "test");
+        boolean isOKForRepair = projectScanner.testBuild(build);
```

Comment:
"isOKForBears"? Other occurrences below too.

Reply:
Here I'm tested with repair mode so it's really "isOKForRepair". I changed below when using "bears" mode.